theme/management/commands/add_quota_avu.py

```diff
@@ -30,6 +34,16 @@ def handle(self, *args, **options):
                             print res.short_id + ' does not have an owner'
                             continue
                     res.set_quota_holder(res.creator, res.creator)
+
+                # Now reset the quota system by setting the AVU 'resetQuotaDir' to anything
+                istorage = res.get_irods_storage()
+                # Add a specific AVU to invoke quota recalculation
+                istorage.session.run("imeta", None, 'set', '-C', res.root_path,
+                                     'resetQuotaDir', '0')
+                # Now remove that AVU to clean up; this is not part of the resource API
+                istorage.session.run("imeta", None, 'rm', '-C', res.root_path,
+                                     'resetQuotaDir', '0')

```

Comment:
@alvacouch @phuongdm @mjstealey Note that this management command only runs once for purpose of migration from a non-quota HydroShare system (i.e., release 1.10) to a quota HydroShare system (i.e., release 1.11), and after release 1.11, this management command will no longer be run since quota holder AVU has already been set initially and will just be carried over from then on. Is this new avu 'resetQuotaDir' used to invoke the PEP micro-service to calculate quota usage for the quota holder? Where is the quota usage for each quota holder stored? I have yet to look at @phuongdm 's microservice code, so not exactly sure what the microservices do for real-time quota calculation. However, is 'resetQuotaDir' AVU really needed here? Is @phuongdm 's microservices going to be wired in for release 1.11? If not, this code addition really serves no purpose, and this management command will no longer be run after release 1.11 since we don't want to reset quota holder after quota system is deployed. An additional management might be needed to achieve your purpose later, but for release 1.11, I believe @mjstealey has specific plans and you will have to coordinate with @mjstealey to come up with best path going forward with quota deployment and further refinement.

Reply:
@aphelionz @hyi @cbcunc @dtarb @phuongdm As requested in Monday's meeting, I have rebased this branch and it now contains only the changes to the script that I intended. These are relatively innocuous changes but the protocol for recalculating quota may have changed. @phuongdm should review. 

Then we should merge if he considers these changes correct.

Reply:
@phuongdm @alvacouch I think we should remove ```resetQuotaDir``` from this add_quota_avu management command. They don't always couple together. This add_quota_avu command only needs to run one time when migrating from a non-quota system to a quota-system initially, which was already done a couple of releases back. In short, this command does not need to be run any more unless we get issues and need to patch things up. However, "resetQuotaDir" needs to be done independently from the add_quota_avu management command. I propose to either add a new "reset_quota_dir" management command in Django, or just simply do it in the hsctl deployment script to directly invoke ```imeta``` command to reset quota dir as needed under proper conditions.

Reply:
I have no idea. With me: resetQuotaDir is a low layer's API 😅. Higher layer could call it by any way as needed